src/main.rs

```diff
@@ -87,3 +79,49 @@ fn parse_args() -> Result<LndCfg, ArgsError> {
 
     Ok(LndCfg::new(address, cert_file, macaroon_file))
 }
+
+/// Sets the onion messaging feature bit (described in this PR: 
+/// https://github.com/lightning/bolts/pull/759/), to signal that we support
+/// onion messaging. This needs to be done every time we start up, because LND
+/// does not currently persist the custom feature bits that are set via the RPC.
+async fn set_feature_bit(mut client: Client) {
+    let update = tonic_lnd::peersrpc::UpdateFeatureAction {
+        action: 0,
+        feature_bit: 38,
+    };
+    let feature_updates = vec![update];
+    let address_updates = vec![];
+
+    let resp = client
+        .peers()
+        .update_node_announcement(tonic_lnd::peersrpc::NodeAnnouncementUpdateRequest {
+            feature_updates: feature_updates,
+            color: String::from(""),
+            alias: String::from(""),
+            address_updates: address_updates,
+        })
+        .await;
+
+    match resp {
+        Ok(_) => {
+	    println!("Now setting the onion messaging feature bit...");
+	},
+        Err(status) => {
+            if !status.message().contains("invalid add action for bit 38, bit is already set") {
```

Comment:
See comment above - to save us from the devilry of string matching, I think it woudl be easier to only call this if we haven't set the feature bit.